components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttConnectionFactory.java

```
@@ -254,35 +170,20 @@ private MqttAsyncClient createMqttAsyncClient() {
                 mqttClient = new MqttAsyncClient(mqttEndpointURL, uniqueClientId);
             }
             log.info("Successfully created to mqtt client");
-        } catch (MqttException e1) {
-            int retryC = 0;
-            while (retryC < retryCount) {
-                retryC++;
-                log.info("Attempting to create mqtt client" + " in " + retryInterval + " ms");
-                try {
-                    Thread.sleep(100);
-                } catch (InterruptedException ignore) {
-                }
-                try {
-                    if (dataStore != null) {
-                        mqttClient = new MqttAsyncClient(mqttEndpointURL, uniqueClientId, dataStore);
-                    } else {
-                        mqttClient = new MqttAsyncClient(mqttEndpointURL, uniqueClientId);
-                    }
-                    log.info("Successfully created to mqtt asynchronous client");
-                } catch (MqttException e) {
-                    log.error("Error while trying to retry", e);
-                }
-            }
-
-            if (mqttClient == null) {
-                log.error("Error while creating the MQTT asynchronous client");
-            } else {
-                log.info("Successfully created to mqtt asynchronous client");
-            }
-
+        } catch (MqttException ex) {
+            log.error("Error while creating the MQTT asynchronous client");
```

Exception trace is swallowed. Please fix.